models/engine/file_storage.py

```diff
@@ -56,11 +61,18 @@ def reload(self):
 
         If the file doesn't exist, no errors or exceptions are raised
         """
+        # TODO: FIND A BETTER WAY OF ORGANISING THIS
```

Comment:
I don't think there can be a way around that. models/ is already initialised globally from the import statement made inside models/__init__.py, so if we move that import statement out of that local scope of the reload() function to the top(globally), we will have an import loop which raises the circular import error. and the cls_map cannot be initialised under reload() if Basemodel class isn't imported and exists in that place. 
Sadly, we can't trick Python here. 
I was only able to validate and use eval() function for dynamic recreation of the class objects which eradicates the declaration of all classes in that place. so no need to have a clss_map dictionary holding all the classes. you can check my follow_up branch, also a few note in the xperiment.py file to explain how I came up with the validation on eval()